packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -742,21 +745,35 @@ class _InterpolationSimulation extends Simulation {
 }
 
 class _RepeatingSimulation extends Simulation {
-  _RepeatingSimulation(this.min, this.max, Duration period)
-    : _periodInSeconds = period.inMicroseconds / Duration.microsecondsPerSecond {
+  _RepeatingSimulation(this.min, this.max, this.reverse, Duration period)
+      : _periodInSeconds = period.inMicroseconds / Duration.microsecondsPerSecond {
     assert(_periodInSeconds > 0.0);
   }
 
   final double min;
   final double max;
+  final bool reverse;
 
   final double _periodInSeconds;
 
+  double _previousT = 0.0;
```

Comment:
That's a very good catch @HansMuller - actually I think it's a bug in the implementation of the  `_RepeatingSimulation` because currently it will always start at the `min` value and go up to `max` based on the elapsed time (and then start over from `min` again). It doesn't care about the controller's initial value at all. 

I have fixed that now by adding the calculated time which it takes to reach the starting value to the `timeInSeconds`.